Send empty strings for deprecated destination fields (#494)
elastic/apm: specs/agents/tracing-spans-destination.md

Rather than opting into the validation, the testing should provide an opt-out if,
for whatever reason, the destination information can't or shouldn't be collected for a particular exit span.

#### `context.destination.service.name`/`context.destination.service.type`

ES field: `span.destination.service.name`/`span.destination.service.type`

The identifier for the destination service.

**Deprecated**

> These fields are deprecated removed from the Elasticsearch documents.

Reviewer: nit:
```suggestion
These fields are deprecated and removed from Elasticsearch documents.
```

> However, we can't just remove them from the intake payloads as they're a required field in the intake API.

Reviewer 2: ```suggestion
However, we can't just remove them from the intake payloads as they're required fields in the intake API.
```

**Value**

Agents MUST NOT manually set these fields.

> Agents MUST NOT offer a non-deprecated public APIs to set them.

Reviewer 2: ```suggestion
Agents MUST NOT offer non-deprecated public APIs to set them.
```
